credentials/src/main/kotlin/SSI.kt

```diff
@@ -228,6 +276,227 @@ public object VerifiableCredential {
+  ): VerifiableCredentialType {
+    val statusListIndexes = prepareCredentialsForStatusList(statusPurpose, issuedCredentials)
+    val bitString = bitstringGeneration(statusListIndexes)
+
+    val claims = mapOf(STATUS_PURPOSE to statusPurpose.toString().lowercase(), ENCODED_LIST to bitString)
+    val credSubject = CredentialSubject.builder()
+      .id(URI.create(statusListCredentialId))
+      .type("StatusList2021")
+      .claims(claims)
+      .build()
+
+    return VerifiableCredentialType.builder()
+      .id(URI.create(statusListCredentialId))
+      .issuer(URI.create(issuer))
+      .issuanceDate(Date())
+      .context(URI.create("https://w3id.org/vc/status-list/2021/v1"))
+      .type("StatusList2021Credential")
+      .credentialSubject(credSubject)
+      .build()
+  }
+
+  /**
+   * Validates if a given credential is part of the status list represented by a `StatusList2021Credential`.
+   *
+   * The function takes in two credentials: one representing the credential to be validated and the other representing the status list credential.
+   * It checks if the given credential's status list index is present in the expanded status list derived from the status list credential.
+   */
+  @Throws(Exception::class)
+  public fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredentialType,
+    statusListCredential: VerifiableCredentialType
+  ): Boolean {
+    val statusListEntryValue: StatusList2021Entry =
+      StatusList2021Entry.fromJsonObject(credentialToValidate.credentialStatus.jsonObject)
+
+    val statusLisCredStatusPurpose: String? =
+      statusListCredential.credentialSubject.jsonObject[STATUS_PURPOSE] as? String?
+
+    requireNotNull(statusListEntryValue.statusPurpose)
+    requireNotNull(statusLisCredStatusPurpose)
+    require(statusListEntryValue.statusPurpose == statusLisCredStatusPurpose)
+
+    val compressedBitstring: String? = statusListCredential.credentialSubject.jsonObject[ENCODED_LIST] as? String?
+
+    requireNotNull(compressedBitstring)
+    require(compressedBitstring.isNotEmpty())
+
+    val credentialIndex = statusListEntryValue.statusListIndex
+    val expandedValues: List<String> = bitstringExpansion(compressedBitstring)
+
+    return expandedValues.any { it == credentialIndex }
+  }
+
+  /**
+   * Validates if a given credential is part of the status list represented by a `StatusList2021Credential`.
+   *
+   * The function takes in a credential representing the one to be validated.
+   * It fetches the status list credential from a URL present in the provided credential.
+   * It then checks if the given credential's status list index is present in the expanded status list derived from the status list credential.
+   */
+  @Throws(Exception::class)
+  public suspend fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredentialType,
+    httpClient: HttpClient = defaultHttpClient() // default HTTP client but can be overridden
```

Comment:
HttpClients should be closed in order to free up connection resources. What do you think would be a good approach here?

Reply:
Updated this!
I handle the closing if its the default client, otherwise if a user defines a client they will be the ones expected to close it